modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcClientTimeZoneTest.java

```diff
@@ -0,0 +1,226 @@
+            stmt.execute("CREATE TABLE test(id INT PRIMARY KEY, ts TIMESTAMP, ts_tz TIMESTAMP WITH LOCAL TIME ZONE)");
+        }
+    }
+
+    @BeforeEach
+    void saveTimeZoneAndClearTable() throws SQLException {
+        origin = TimeZone.getDefault().toZoneId();
+
+        stmt.execute("DELETE FROM test");
+    }
+
+    @AfterEach
+    void restoreTimezone() {
+        ZoneId current = TimeZone.getDefault().toZoneId();
+
+        if (!Objects.equals(origin, current)) {
+            TimeZone.setDefault(TimeZone.getTimeZone(origin));
+        }
+    }
+
+    /** Ensures that the default JVM time zone is passed to the server. */
+    @Test
+    public void jvmTimeZonePassedToServer() throws SQLException {
+        ZoneId serverTimezone = TimeZone.getTimeZone("GMT+1").toZoneId();
+
+        // Client time zone.
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT+02:00"));
+
+        withNewConnection(URL, stmt -> {
+            // Set server timezone.
+            TimeZone.setDefault(TimeZone.getTimeZone(serverTimezone));
+
+            stmt.executeUpdate(format("INSERT INTO test VALUES(0, '{}', '{}')", TIMESTAMP_STR, TIMESTAMP_STR));
+
+            validateSingleRow("SELECT ts::VARCHAR, ts_tz::VARCHAR FROM test", stmt,
+                    "1970-01-01 00:00:00", "1970-01-01 00:00:00 GMT+02:00");
+        });
+
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT+03:00"));
+
+        withNewConnection(URL, stmt -> {
+            // Set server timezone.
+            TimeZone.setDefault(TimeZone.getTimeZone(serverTimezone));
+
+            validateSingleRow("SELECT ts::VARCHAR, ts_tz::VARCHAR FROM test", stmt,
+                    "1970-01-01 00:00:00", "1970-01-01 01:00:00 GMT+03:00");
+        });
+    }
+
+    /**
+     * Ensures that session time zone can be changed using
+     * connection property {@link ConnectionProperties#setConnectionTimeZone(ZoneId)}.
+     */
+    @Test
+    public void timeZoneCanBeSetUsingProperty() throws SQLException {
+        String originTimeZone = TimeZone.getDefault().getID();
+
+        {
+            String timeZone = "GMT+02:00";
+
+            withNewConnection(URL + "?connectionTimeZone=" + timeZone, stmt -> {
+                stmt.executeUpdate(format("INSERT INTO test VALUES(0, '{}', '{}')", TIMESTAMP_STR, TIMESTAMP_STR));
+
+                validateSingleRow("SELECT ts::VARCHAR, ts_tz::VARCHAR FROM test", stmt,
+                        "1970-01-01 00:00:00", "1970-01-01 00:00:00 " + timeZone);
+            });
+        }
+
+        {
+            String timeZone = "GMT+03:00";
+
+            withNewConnection(URL + "?connectionTimeZone=" + timeZone, stmt -> {
+                validateSingleRow("SELECT ts::VARCHAR, ts_tz::VARCHAR FROM test", stmt,
+                        "1970-01-01 00:00:00", "1970-01-01 01:00:00 " + timeZone);
+            });
+        }
+
+        {
+            String timeZone = "invalid/timezone";
+
+            assertThrows(
+                    ZoneRulesException.class,
```

Comment:
do we need to wrap such exception into appropriate sqlexception ?

Reply:
Thanks, I missed this one, wrapped in a standard jdbc SQLException (like other jdbc property validation errors).